### PR TITLE
[MRG] Shaving off some seconds from the tests

### LINKED ISSUE #449: Speed up tests, part deux
We get intermittent travis failures because the tests take too long to run 😢 

Short of just disabling all tests marked as "slow", what else can we do? Reduce number of iterations in minimizer loops?

### PR DESCRIPTION
**Pull request by betatim**
Fixes #449 

Trying to cut down on the combinatorics mostly.

### CODE REVIEW

**Comment by betatim** [comment]
This mostly cuts down the `n_calls` parameters to speed things up.

A few tests I also switched from all `*_minimize` functions times all acquisition functions to just all minimize functions as the parametrisation. I think this should be fine but if someone else could take a look that would be great.

File: skopt/tests/test_common.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def test_init_points_and_models(n_random_starts, optimizer_func):
     # starts (no y0 in this case)
     space = [(-5.0, 10.0), (0.0, 15.0)]
     x0 = [[1, 2], [3, 4], [5, 6]]
-    n_calls = 8
+    n_calls = 7
```
Comment (MechCoder) [inline skopt/tests/test_common.py]: I think you can still reduce this to `n_calls=6`?

File: skopt/tests/test_common.py [hunk 2]
```diff
@@ -244,7 +244,7 @@ def test_init_points_and_models(n_random_starts, optimizer_func):
 def test_init_vals(n_random_starts, optimizer_func):
     space = [(-5.0, 10.0), (0.0, 15.0)]
     x0 = [[1, 2], [3, 4], [5, 6]]
-    n_calls = 10
+    n_calls = len(x0) + n_random_starts
```
Comment (MechCoder) [inline skopt/tests/test_common.py]: I would suggest this to reduce this rather to check the case where this exceeds len(x0) + n_random_starts

**Comment by MechCoder** [comment]
Just a couple of minor comments. We should probably look at the difference in the speed of the tests before merging from now on

File: skopt/tests/test_common.py (hunk 2 above)
Comment (betatim) [inline skopt/tests/test_common.py]: So you mean `n_calls = len(x0) + n_random_starts + 1`?

File: skopt/tests/test_common.py (hunk 1 above)
Comment (betatim) [inline skopt/tests/test_common.py]: At the maximum we have `n_random_starts=4` and three initial points so we need seven calls. I think. At least we get an error if I reduce it more :-/

**Comment by glouppe** [comment]
Let's merge this! Travis really is too slow :(

**Comment by betatim** [comment]
image
 Let's enjoy it while it lasts.

**Comment by MechCoder** [comment]
lgtm

**Pull request merged**